pkg/controllers/ttl/controller.go

```diff
@@ -117,6 +118,34 @@ func (c *controller) deregisterEventHandlers() {
 	c.logger.V(3).Info("deregistered event handlers")
 }
 
+// Function to determine the deletion propagation policy
+func (c *controller) determinePropagationPolicy(metaObj metav1.Object, logger logr.Logger) *metav1.DeletionPropagation {
```

Comment:
no need for a method here

Reply:
Sure

Reply:
Could you clarify why we should avoid creating a separate method for determining the deletion propagation policy?

Reply:
I mean a simple func is enough, it doesn't need the controller

Reply:
Okay got it.